### ISSUE TITLE
TextEdit is not applied correctly 

### ISSUE THREAD

**Issue by predragnikolic** [sublimelsp/LSP#957]
![output](https://user-images.githubusercontent.com/22029477/78840351-c8733600-79fa-11ea-82c1-8bb265d2299c.gif)

```
            # this region was valid a few view.change_count() moments back ...
            edit_region = range_to_region(Range.from_lsp(text_edit['range']), self.view)
            # ... but this brings it to the present.
  mutate >> edit_region = self.view.transform_region_from(edit_region, item["change_id"])
```

We are probably not passing the correct region to the transform_region_from.
We should also not mutate the TextEdit from the server. (the region returned from `transform_region_from` is not the same as the text edit region returned from the server)

I think that we need to grab a snapshot(something like restore_lines, or something different).

**Comment by rwols** [comment]
It's possibly a bug in `transform_region_from`, or a misunderstanding on how that function is supposed to be used.

This is using `transform_region_from` (it should remove `u>`, but doesn't):
![Peek 2020-04-09 18-55](https://user-images.githubusercontent.com/2431823/78925357-9665e080-7a9b-11ea-8211-56565f3ae458.gif)

This skips `transform_region_from` and uses the historic region as-is:
![Peek 2020-04-09 18-57](https://user-images.githubusercontent.com/2431823/78925418-ada4ce00-7a9b-11ea-9ffe-527408efef09.gif)

**Comment by rwols** [comment]
I believe the changes in #962 inadvertently fixed your issue too:

![Peek 2020-04-11 15-17](https://user-images.githubusercontent.com/2431823/79044811-a39fde00-7c07-11ea-9be6-738df8a1f8cb.gif)

The last remaining thing to do is to fix up the auto complete triggers.

**Comment by rwols** [comment]
Fixed by #962
